blackjax/base.py

```diff
@@ -0,0 +1,98 @@
+from typing import NamedTuple, Tuple
+
+from typing_extensions import Protocol
+
+from .types import PRNGKey, PyTree
+
+Position = PyTree
+State = NamedTuple
+Info = NamedTuple
+
+
+class InitFn(Protocol):
```

Comment:
Why typing Protocal here and not below?

Reply:
That's an excellent question, I forgot and I think that's why I'm getting type errors.